### DISCUSSION TITLE
What kind of error is E_OUTDATED_TERM(-3071)?

### DISCUSSION THREAD

**Discussion by happymoment-s** [vesoft-inc/nebula#4275] [Q&A] [answered]
Hello.
`-1005:Storage Error: part: 17, error: E_OUTDATED_TERM(-3071).`
When does the `-3071` error occur?
What does `Out-of-date heartbeat received from the old leader (the new leader has been elected)` mean? 
(https://docs.nebula-graph.io/3.1.0/20.appendix/error-code/)

[Environment]
nebula 3.0.0 (com.vesoft:client:3.0.0)
spring boot/java

[Query]
```
DELETE VERTEX "a" WITH EDGE
DELETE VERTEX "b" WITH EDGE
DELETE EDGE e_edge "a"->"b"
```
result: -1005:Storage Error: part: 17, error: E_OUTDATED_TERM(-3071).

```
DELETE VERTEX "c" WITH EDGE
DELETE VERTEX "d" WITH EDGE
DELETE EDGE e_edge "c"->"d"
```
result: ok

This happens every time I delete a specific vertex or I delete a related edge. It doesn't happen if I delete another vertex or edge.
Whether the vertex does not exist or exists, an error always occurs at a specific vertex.
For reference, I recently changed from 2.6.2 to 3.0.0. It didn't happen in 2.6.2. Is this a new error in 3.0.0? Please check.

Let me know if you need additional information. :D

**Comment by wey-gu** [comment, marked as answer]
I think this is related to TOSS(edge transaction on the storage side), you enabled the experimental feature, right? Thus TOSS was used in the edge deletion/add, which ensures edge deletion/add with a transaction guarantee.

In TOSS/chained edge deletion, term id was checked towards the term id in metaD before committing the write request to enable transaction(In nebula, an edge was persisted in multiple Key-Value records, this was to enable its atomicity).

**Reply by happymoment-s** [reply]
Thank you for answer! Let's turn off the experimental feature and test again. :D

**Reply by wey-gu** [reply]
Any time @happymoment-s :D, when this error is encountered, I think a retry(of the edge deletion) from the client-side should be fine, too(rather than turning TOSS off), thanks!
